app/src/main/java/io/neurolab/ConfigUtils.java

```diff
@@ -0,0 +1,78 @@
+package io.neurolab;
+
+import android.content.Context;
+import com.google.gson.Gson;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class ConfigUtils {
+
+    private static final String CONFIG_FILE_NAME = "config.json";
+
+    public static ConfigurationSettings loadSettingsConfig(Context context){
+        File configFile = new File(context.getFilesDir(), CONFIG_FILE_NAME);
+        if(!configFile.isFile()){
+            return createConfigFile(configFile);
```

Comment:
Just return a new instance of `ConfigurationSettings`

Reply:
@iamareebjamal so shall I do something like this 

    if(!configFile.isFile()){
        createConfigFile(configFile);
        return new ConfigurationSettings();
    }
I am asking this because createConfigFile() function has to be called anyways as we have to create the file for the first time. I actually changed the createConfigFile() methods signature to return a new ConfigurationSettings object itself. We can even keep them separate as written the code above. 
Correct me if I am wrong and let me know if I am understanding correctly or not.

Reply:
Yes you should just return this

` createConfigFile() function has to be called anyways as we have to create the file for the first time`

Why? 

Just create it when user changes a setting